docs/specification-ligo.md

```diff
@@ -156,19 +156,19 @@ Each address can have any number of operators and be an operator of any number o
 ## Proposals
 
 Everyone can make a new proposal, however, you have to freeze some tokens for that.
-The proposer specifies how many tokens they want to freeze and this value is checked by
-the contract according to its compile-time configuration.
-If this value is accepted and the proposer has enough unfrozen tokens, these unfrozen tokens
-are frozen and voting starts.
+The proposer specifies how many frozen tokens they want to stake and this value is checked by
+the contract according to its compile-time configuration. Also tokens frozen in a voting period
+can only be used in the next period and further. This applies to proposal creation as well as to
+the `unfreeze` operation.
 
 Proposals are identified by a key which is a `bytes` value computed via Blake2B hashing function of a
 pair of propose entrypoint params and the proposer address.
 
 ## Voting
 
-Once a proposal is submitted, everyone can vote on it as long as they have enough unfrozen tokens.
-One unfrozen token is required for one vote.
-The tokens are frozen for the duration of voting.
+Once a proposal is submitted, everyone can vote on it as long as they have enough frozen tokens to stake.
+One frozen token is required for one vote. Here also, only tokens frozen during the previous periods can
+be used to cast votes.
 Voting period is specified for the whole smart contract and can be updated by the administrator; on update, the existing proposals are also affected.
 It's possible to vote positively or negatively.
 After the voting ends, the contract is "flushed" by calling a dedicated entrypoint.
```

Comment:
(writing this here because I cannot comment below)

3 new errors need to be added to the table below: `NOT_PROPOSING_PERIOD`, `NOT_ENOUGH_FROZEN_TOKENS`, and `NOT_ENOUGH_STAKED_TOKENS`

Reply:
Have added.